File: 14UI/14UI.py
import lcm
import time
import socket
import json

from lilylcm import L06Depth
from lilylcm import L07Humidity
from lilylcm import L08Tempurature
from lilylcm import L14LEDs
from lilylcm import L16ChargerCommand
from lilylcm import L19DockCommand

lc = lcm.LCM()

IP = input("What is the servers IP address? ")

HOST = IP
PORT = 8888
s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
s.connect((HOST, PORT))
dic = {'Hum': 0, 'Temp': 0, 'Dep': 0, 'Sat': 0, 'Charge': 0, 'Mag': 0}

def my_handler(channel, data):
  if channel == "POD_Depth":
    datamsg = L06Depth.decode(data)
    dic['Dep'] = datamsg.depth
  elif channel == "09I2C_HUMIDITY":
    datamsg = L07Humidity.decode(data)
    dic['Hum'] = datamsg.humidity
  elif channel == "09I2C_TEMP":
    datamsg = L08Tempurature.decode(data)
    dic['Temp'] = datamsg.tempurature
  elif channel == "POD_LED":
    datamsg = L14LEDs.decode(data)
    dic['Sat'] = datamsg.switchOn
  elif channel == "POD_Charge":
    datamsg = L16ChargerCommad.decode(data)
    dic['Charge'] = datamsg.targetState
  elif channel == "POD_Magnet":
    datamsg = L19DockCommand.decode(data)
    dic['Mag'] = datamsg.switchOn
  else:
    pass
  
  s.send(json.dumps(dic))

subDep = lc.subscribe("POD_Depth", my_handler)
subHum = lc.subscribe("09I2C_HUMIDITY", my_handler)
subTemp = lc.subscribe("09I2C_TEMP", my_handler)
subSat = lc.subscribe("POD_LED", my_handler) 
subChar = lc.subscribe("POD_Charge", my_handler)
subMag = lc.subscribe("POD_Magnet", my_handler)

try:
  while True:
    lc.handle()
    time.sleep(0.05)
except KeyboardInterrupt:
  lc.unsubscribe(sub)
  s.close()

File: 14UI/14UI.cpp
#include <stdio.h>

#include <lcm/lcm-cpp.hpp>
#include "lilylcm/L06Depth.hpp"
#include "lilylcm/L07Humidity.hpp"
#include "lilylcm/L08Tempurature.hpp"
#include "lilylcm/L14LEDs.hpp"
#include "lilylcm/L16ChargerCommand.hpp"
#include "lilylcm/L19DockCommand.hpp"

class Handler 
{
    public:
        ~Handler() {}

        void handleMessage(const lcm::ReceiveBuffer* rbuf,
                const std::string& chan, 
                const lilylcm* msg)
        {
            int i;
            printf("Received message on channel \"%s\":\n", chan.c_str());
            printf("  timestamp   = %lld\n", (long long)msg->timestamp);
            printf("  position    = (%f, %f, %f)\n",
                    msg->position[0], msg->position[1], msg->position[2]);
            printf("  orientation = (%f, %f, %f, %f)\n",
                    msg->orientation[0], msg->orientation[1], 
                    msg->orientation[2], msg->orientation[3]);
            printf("  ranges:");
            for(i = 0; i < msg->num_ranges; i++)
                printf(" %d", msg->ranges[i]);
            printf("\n");
            printf("  name        = '%s'\n", msg->name.c_str());
            printf("  enabled     = %d\n", msg->enabled);
        }
};

int main(int argc, char** argv)
{
    lcm::LCM lcm;

    if(!lcm.good())
        return 1;

    Handler handlerObject;
    lcm.subscribe("POD_Depth", &Handler::handleMessage, &handlerObject);
    lcm.subscribe("POD_Humidity", &Handler::handleMessage, &handlerObject);
    lcm.subscribe("POD_Tempurature", &Handler::handleMessage, &handlerObject);
    lcm.subscribe("POD_LED", &Handler::handleMessage, &handlerObject);
    lcm.subscribe("POD_Charge", &Handler::handleMessage, &handlerObject);
    lcm.subscribe("POD_Magnet", &Handler::handleMessage, &handlerObject);

    while(0 == lcm.handle());

    return 0;
}

